rebased the branch, seems there are no additional lang dependent failures but I cannot say for sure as its still NOT building:

`
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project keycloak-saml-core: Compilation failure: Compilation failure:
[ERROR] /Users/rlfnb/workspace/keycloak/saml-core/src/main/java/org/keycloak/saml/processing/core/util/XMLSignatureUtil.java:[757,42] incompatible types: java.util.List<java.lang.Object> cannot be converted to java.util.List<? extends javax.xml.crypto.XMLStructure>
[ERROR] /Users/rlfnb/workspace/keycloak/saml-core/src/main/java/org/keycloak/saml/processing/api/util/KeyInfoTools.java:[51,70] incompatible types: java.util.List<javax.xml.crypto.XMLStructure> cannot be converted to java.lang.Iterable<java.lang.Object>
[ERROR] /Users/rlfnb/workspace/keycloak/saml-core/src/main/java/org/keycloak/saml/processing/api/util/KeyInfoTools.java:[55,70] incompatible types: java.util.List<javax.xml.crypto.XMLStructure> cannot be converted to java.lang.Iterable<java.lang.Object>
[ERROR] /Users/rlfnb/workspace/keycloak/saml-core/src/main/java/org/keycloak/saml/processing/api/util/KeyInfoTools.java:[60,58] incompatible types: java.util.List<capture#1 of ?> cannot be converted to java.lang.Iterable<java.lang.Object>
`
started maven as below:
`
$ mvn -X clean package
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T18:41:47+02:00)
Maven home: /Users/rlfnb/java/apache-maven-3.3.9
Java version: 10.0.1, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.10.0.1/Contents/Home
Default locale: de_DE, platform encoding: UTF-8
OS name: "mac os x", version: "10.13.4", arch: "x86_64", family: "mac"
`

Any obvious idea so far? Would you merge the first fix into master or should we try to resolve the other issue as well?

--- a/saml-core/src/test/java/org/keycloak/saml/processing/core/parsers/saml/SAMLParserTest.java
+++ b/saml-core/src/test/java/org/keycloak/saml/processing/core/parsers/saml/SAMLParserTest.java
@@ -708,7 +708,8 @@ public class SAMLParserTest {
     @Test
     public void testInvalidEndElement() throws Exception {
         thrown.expect(ParsingException.class);
-        thrown.expectMessage(containsString("The element type \"NameIDFormat\" must be terminated by the matching end-tag \"</NameIDFormat>\"."));
+        // see KEYCLOAK-7444 
+        thrown.expectMessage(containsString("NameIDFormat"));
 
         assertParsed("saml20-entity-descriptor-idp-invalid-end-element.xml", EntityDescriptorType.class);
     }

